### ISSUE TITLE
(cd examples; bazel build //...) fails with bazel 0.23.0: conflicting actions

### ISSUE THREAD

**Issue by smparkes** [grpc/grpc-java#5412]
### What version of gRPC are you using?

Seeing this on v0.19.0 and master.

### What did you expect to see?

Successful build.

This is what I'm seeing:
```
~/u/grpc-java git:master ❯❯❯ (cd examples; bazel build //...)
ERROR: file 'libroute_guide_java_grpc-src.jar' is generated by these conflicting actions:
Label: //:route_guide_java_grpc
RuleClass: _java_rpc_library rule
Configuration: 58aef4e3a9f35f464a91ebf395e1ced2
Mnemonic: JavaSourceJar, SkylarkAction
Action key: 98dee0f27646ce92498e66fcdb2c93ce, b6319fcef16694a80853bbbe03601580
Progress message: Building source jar libroute_guide_java_grpc-src.jar, SkylarkAction libroute_guide_java_grpc-src.jar
PrimaryInput: File:[[<execution_root>]bazel-out/darwin-fastbuild/bin]libroute_guide_java_grpc-src.jar, File:[/Users/smparkes/upstream/grpc-java/examples[source]]src/main/proto/route_guide.proto
PrimaryOutput: File:[[<execution_root>]bazel-out/darwin-fastbuild/bin]libroute_guide_java_grpc-src.jar
ERROR: Analysis of target '//:route_guide_java_grpc' failed; build aborted: for libroute_guide_java_grpc-src.jar, previous action: action 'SkylarkAction libroute_guide_java_grpc-src.jar', attempted action: action 'Building source jar libroute_guide_java_grpc-src.jar'
INFO: Elapsed time: 0.359s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
```

**Comment by drigz** [comment]
See #5395.

**Comment by smparkes** [comment]
Thanks, @drigz !

**Comment by smparkes** [comment]
Hmmm, since @carl-mastrangelo assigned this maybe I shouldn't have closed it ...

**Comment by zhangkun83** [comment]
We should keep it open until #5395 or its equivalent is merged.

**Comment by ejona86** [comment]
And as you can see for #5395, any fix for 0.23 will break some users of 0.22 and before.

**Comment by ejona86** [comment]
#5395 is merged to master which requires Bazel 0.23. #5429 will be the change for the release branch to retain Bazel 0.22 compatibility. But if you are using the -src.jar output of the rule you will need to use `--incompatible_generate_javacommon_source_jar` in Bazel 0.22 or upgrade to Bazel 0.23.

**Comment by ejona86** [comment]
Marking resolved since #5429 is now in.

**Comment by smparkes** [comment]
Thanks, @ejona86!
